nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/logging/repository/StandardLogRepository.java

```diff
@@ -46,43 +50,60 @@
     private volatile ComponentLog componentLogger;
 
     @Override
-    public void addLogMessage(final LogLevel level, final String message) {
-        addLogMessage(level, message, (Throwable) null);
-    }
-
-    @Override
-    public void addLogMessage(final LogLevel level, final String message, final Throwable t) {
-        final LogMessage logMessage = new LogMessage(System.currentTimeMillis(), level, message, t);
+    public void addLogMessage(LogMessage logMessage) {
+        LogLevel logLevel = logMessage.getLogLevel();
 
-        final Collection<LogObserver> logObservers = observers.get(level);
+        final Collection<LogObserver> logObservers = observers.get(logLevel);
         if (logObservers != null) {
             for (LogObserver observer : logObservers) {
                 try {
                     observer.onLogMessage(logMessage);
-                } catch (final Throwable observerThrowable) {
+                } catch (final Exception observerThrowable) {
                     logger.error("Failed to pass log message to Observer {} due to {}", observer, observerThrowable.toString());
                 }
             }
         }
+
     }
 
     @Override
     public void addLogMessage(final LogLevel level, final String format, final Object[] params) {
         replaceThrowablesWithMessage(params);
+        final Optional<String> flowFileUUID = getFlowFileUUIDFromObjects(params);
         final String formattedMessage = MessageFormatter.arrayFormat(format, params).getMessage();
-        addLogMessage(level, formattedMessage);
+        final LogMessage logMessage = new LogMessage.Builder(System.currentTimeMillis(), level)
+                .message(formattedMessage)
+                .flowFileUUID(flowFileUUID.orElse(null))
+                .createLogMessage();
+        addLogMessage(logMessage);
     }
 
     @Override
     public void addLogMessage(final LogLevel level, final String format, final Object[] params, final Throwable t) {
         replaceThrowablesWithMessage(params);
+        final Optional<String> flowFileUUID = getFlowFileUUIDFromObjects(params);
         final String formattedMessage = MessageFormatter.arrayFormat(format, params, t).getMessage();
-        addLogMessage(level, formattedMessage, t);
+        final LogMessage logMessage = new LogMessage.Builder(System.currentTimeMillis(), level)
+                .message(formattedMessage)
+                .throwable(t)
+                .flowFileUUID(flowFileUUID.orElse(null))
+                .createLogMessage();
+        addLogMessage(logMessage);
+    }
+
+    private Optional<String> getFlowFileUUIDFromObjects(Object[] params) {
```

Comment:
Might make sense to name this method `getFirstFlowFileUuidFromObjects` for clarity - it's definitely possible that there could be multiple FlowFile objects. Or, it might make more sense to return `Optional.empty()` if there's more than one FlowFile, as we really don't know which FlowFile the log message is really focused on. For example, the log message `Created {} from {} but failed to send {}` may take 3 FlowFiles are arguments. The first one would be the most important FlowFile there. But for a message like `Created 3 FlowFiles from {} but {} could not be sent`, the most important FlowFile for the log message is the second. Can't really know which is the main theme of the log message.

Reply:
That's a good idea!